pkg/jsonAPI/jsonapi.go

```diff
@@ -0,0 +1,514 @@
+
+	latestDays := fmt.Sprintf("latest%dDays", endDay)
+	prevDays := "prev7Days"
+
+	allPrev := resultPrev["all"]
+
+	var topFailingTests []FailingTestBug
+
+	for _, test := range topFailingTestsWithoutBug {
+		encodedTestName := url.QueryEscape(regexp.QuoteMeta(test.Name))
+
+		testLink := fmt.Sprintf("https://search.svc.ci.openshift.org/?maxAge=168h&context=1&type=bug%%2Bjunit&name=&maxMatches=5&maxBytes=20971520&groupBy=job&search=%s", encodedTestName)
+		testPrev := getPrevTest(test.Name, allPrev.TestResults)
+
+		var failedTestWithoutBug FailingTestBug
+
+		if testPrev != nil {
+
+			failedTestWithoutBug = FailingTestBug{
+				Name: test.Name,
+				Url:  testLink,
+				PassRates: map[string]PassRate{
+					latestDays: PassRate{
+						Percentage: test.PassPercentage,
+						Runs:       test.Successes + test.Failures,
+					},
+					prevDays: PassRate{
+						Percentage: testPrev.PassPercentage,
+						Runs:       testPrev.Successes + testPrev.Failures,
+					},
+				},
+			}
+
+		} else {
+			failedTestWithoutBug = FailingTestBug{
+				Name: test.Name,
+				Url:  testLink,
+				PassRates: map[string]PassRate{
+					latestDays: PassRate{
+						Percentage: test.PassPercentage,
+						Runs:       test.Successes + test.Failures,
+					},
+				},
+			}
+		}
+
+		topFailingTests = append(topFailingTests, failedTestWithoutBug)
+	}
+	return topFailingTests
+}
+
+func jsonSummaryJobPassRatesByJobName(report, reportPrev util.TestReport, endDay, jobTestCount int) []PassRatesByJobName {
+	latestDays := fmt.Sprintf("latest%dDays", endDay)
+	prevDays := "prev7Days"
+	jobRunsByName := util.SummarizeJobsByName(report)
+	jobRunsByNamePrev := util.SummarizeJobsByName(reportPrev)
+
+	var passRatesSlice []PassRatesByJobName
+
+	for _, v := range jobRunsByName {
+		prev := getPrevJob(v.Name, jobRunsByNamePrev)
+
+		var newJobPassRate PassRatesByJobName
+
+		if prev != nil {
+			newJobPassRate = PassRatesByJobName{
+				Name: v.Name,
+				Url:  v.TestGridUrl,
+				PassRates: map[string]PassRate{
+					latestDays: PassRate{
+						Percentage:          v.PassPercentage,
+						ProjectedPercentage: v.PassPercentageWithKnownFailures,
+						Runs:                v.Successes + v.Failures,
+					},
+					prevDays: PassRate{
+						Percentage:          prev.PassPercentage,
+						ProjectedPercentage: prev.PassPercentageWithKnownFailures,
+						Runs:                prev.Successes + prev.Failures,
+					},
+				},
+			}
+		} else {
+			newJobPassRate = PassRatesByJobName{
+				Name: v.Name,
+				Url:  v.TestGridUrl,
+				PassRates: map[string]PassRate{
+					latestDays: PassRate{
+						Percentage:          v.PassPercentage,
+						ProjectedPercentage: v.PassPercentageWithKnownFailures,
+						Runs:                v.Successes + v.Failures,
+					},
+				},
+			}
+		}
+
+		// FIXME: not sure if I need this
+		count := jobTestCount
```

Comment:
you don't, if you're trying to return all the tests that failed for the job, not a limited set.

you can remove it just like you removed it from the ByPlatform results above.

It does mean you're going to potentially return a *lot* of data (every test that failed across every job, multiplied by 2 since you're also doing it in the byplatform results).

So you may want to look at how big a json this is producing and evaluate whether returning all the tests is really viable or useful.

Reply:
(if you don't remove it, you should also considering adding additionalMatches back in so the api can inform the caller of how many results were skipped)

Reply:
for now, I think I'm going to remove it, because I think it I could add a feature to only request specific portions so that less data is returned

Reply:
I agree that there is too much data at the moment though